android/app/src/main/java/com/cBreez/client/BreezForegroundService.kt

```diff
@@ -0,0 +1,148 @@
+package com.cBreez.client
+
+import android.app.Service
+import android.content.Intent
+import android.os.Build
+import android.os.Handler
+import android.os.IBinder
+import android.os.Looper
+import breez_sdk.BlockingBreezServices
+import breez_sdk.BreezEvent
+import breez_sdk.EventListener
+import com.cBreez.client.BreezNotificationHelper.Companion.notifyForegroundService
+import com.cBreez.client.BreezNotificationHelper.Companion.notifyPaymentReceived
+import com.cBreez.client.BreezNotificationHelper.Companion.registerNotificationChannels
+import com.cBreez.client.BreezSdkConnector.Companion.connectSDK
+import com.cBreez.client.Constants.EXTRA_REMOTE_MESSAGE
+import com.cBreez.client.Constants.NOTIFICATION_ID_FOREGROUND_SERVICE
+import com.google.firebase.messaging.RemoteMessage
+import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.launch
+import org.tinylog.kotlin.Logger
+
+class BreezForegroundService : Service() {
+    companion object {
+        private const val TAG = "BreezForegroundService"
+        private const val SHUTDOWN_DELAY_MS = 60 * 1000L // 60 seconds
+    }
+
+    // SDK events listener
+    inner class SDKListener : EventListener {
+        override fun onEvent(e: BreezEvent) {
+            Logger.tag(TAG).info { "Received event $e" }
+            when (e) {
+                is BreezEvent.InvoicePaid -> {
+                    val pd = e.details
+                    handleReceivedPayment(pd.bolt11, pd.paymentHash, pd.payment?.amountMsat)
+
+                    // Push back shutdown by SHUTDOWN_DELAY_MS in case we'll receive more payments
+                    pushbackShutdown()
```

Comment:
I don't think we need that. It makes the foreground service keep running in the task bar.
Also we need to wait for sync after InvoicePaid, see the corresponding implementation in iOS.

Reply:
Done [Shutdown service after receiving synced event for received payment](https://github.com/breez/c-breez/pull/748/commits/5feeb18c39e1cd8f48d75183afeff74706e00373)

made a little change later to only log if payment is received https://github.com/breez/c-breez/pull/748/commits/79d7566aebf9ea00d6c65143b6a019fbb72f7a5c